### PR TITLE
Misc Atom Improvements.

### PR DESCRIPTION
**Pull request by antonmic**
Added ifdefs to disable shadow code when included in files that define ENABLE_SHADOWS to be 0, allowing such shaders to compile. Added option for RasterPass to set its viewport/scissor based on its output. Added RasterPass draw item count to ImGuiPassTree visualization to help debug how many draw items each RasterPass is receiving.

### CODE REVIEW

**Review by AMZN-alexpete — APPROVED** [review]

File: Gems/Atom/RPI/Code/Source/RPI.Public/Pass/RasterPass.cpp [hunk 1]
```diff
@@ -130,15 +131,25 @@ namespace AZ
...
         {
-            if (!m_overrideScissorSate)
+            if (m_viewportAndScissorTargetOutputIndex >= 0)
             {
-                m_scissorState = params.m_scissorState;
+                PassAttachmentBinding& target = GetOutputBinding(m_viewportAndScissorTargetOutputIndex);
+                u32 targetWidth = target.GetAttachment()->m_descriptor.m_image.m_size.m_width;
+                u32 targetHeight = target.GetAttachment()->m_descriptor.m_image.m_size.m_height;
+                m_scissorState = RHI::Scissor(0, 0, targetWidth, targetHeight);
+                m_viewportState = RHI::Viewport(0, (float)targetWidth, 0, (float)targetHeight);
```
Comment (moudgils) [inline Gems/Atom/RPI/Code/Source/RPI.Public/Pass/RasterPass.cpp:140]: static_cast<float>

File: Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Pass/RasterPassData.h [hunk 2]
```diff
@@ -39,6 +39,10 @@ namespace AZ
             RHI::Scissor m_overrideScissor = RHI::Scissor::CreateNull();
 
             RHI::DrawListSortType m_drawListSortType = RHI::DrawListSortType::KeyThenDepth;
+
+            // Forces viewport and scissor to match width/height of output image at specified index.
+            // Does nothing if index is negative.
+            s32 m_viewportAndScissorTargetOutputIndex = -1;
```
Comment (moudgils) [inline Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Pass/RasterPassData.h:45]: Why use a signed int instead of a boolean?

Comment (antonmic) [reply Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Pass/RasterPassData.h:45]: To specify which of the outputs' dimension to use for the viewport/scissor
-1: Don't do anything
0: Use output[0] for viewport height/width
1: Use output[1] for viewport height/width
etc.

File: Gems/Atom/RPI/Code/Source/RPI.Public/Pass/RasterPass.cpp (hunk 1 above)
Comment (antonmic) [reply Gems/Atom/RPI/Code/Source/RPI.Public/Pass/RasterPass.cpp:140]: Will fix

Comment (antonmic) [reply Gems/Atom/RPI/Code/Source/RPI.Public/Pass/RasterPass.cpp]: Fixed

File: Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Pass/RasterPassData.h (hunk 2 above)
Comment (moudgils) [reply Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Pass/RasterPassData.h:45]: oh I somehow missed that code.

**Pull request merged**